src/pods/generate/SubfolderResourcesGenerator.ts

```diff
@@ -0,0 +1,112 @@
+import type { ResourceIdentifier } from '../../http/representation/ResourceIdentifier';
+import { joinFilePath, resolveAssetPath } from '../../util/PathUtil';
+import type { Resource } from './ResourcesGenerator';
+import type { TemplatedResourcesGenerator } from './TemplatedResourcesGenerator';
+import Dict = NodeJS.Dict;
+
+/**
+ * Generates all resources found in specific subfolders of the given template folder.
+ * In case the same resource is defined in several subfolders,
+ * the data of the last subfolder in the list will be used.
+ *
+ * The results of all the subfolders will be merged so the end result is still a sorted stream of identifiers.
+ *
+ * One of the main use cases for this class is so template resources can be in a separate folder
+ * than their corresponding authorization resources,
+ * allowing for authorization-independent templates.
+ */
+export class SubfolderResourcesGenerator implements TemplatedResourcesGenerator {
```

Comment:
I was a bit surprised with this mechanism.

I thought that it would be like:
1. `templates/pod/documents`
2. `templates/pod/wac`
3. `templates/pod/acp`

And then that either:
- there would be generator that is an aggregate of "generator for 1" and "generator for 2 (or 3)"
- a generator could take multiple paths ("to https://pod.example.org/, generate the union of `templates/one/documents` and `templates/one/wac`)

So I don't understand on a high level why there is a need for a non-trivial subfolder class.

Reply:
Summary after discussion: create separate class/utility function to do a sorted merge of the iterables and use that in the SubfolderResourcesGenerator so that class focusses on what is specific for the subfolders.